.travis.yml

```diff
@@ -32,20 +33,33 @@ jobs:
           mkdir -p $HOME/docker && docker images -a --filter='dangling=false' --format '{{.Repository}}:{{.Tag}} {{.ID}}'
           | xargs -n 2 -t sh -c 'test -e $HOME/docker/$1.tar.gz || docker save $0 | gzip -2 > $HOME/docker/$1.tar.gz'
     - stage: danger lint and jest
-      script:
-        - "travis_wait docker-compose run -e DANGER_GITHUB_API_TOKEN=$DANGER_GITHUB_API_TOKEN -e TRAVIS=$TRAVIS -e CI=$CI -e HAS_JOSH_K_SEAL_OF_APPROVAL=$HAS_JOSH_K_SEAL_OF_APPROVAL -e TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST -e TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG test-ci bash -c 'npm run danger && npm run lint && npm run jest'"
-    - stage: wdio
-      script:
-        - "travis_wait docker-compose run test-ci npm run wdio -- --suite suite1"
-    -
-      script:
-        - "travis_wait docker-compose run test-ci npm run wdio -- --suite suite2"
-    -
-      script:
-        - "travis_wait docker-compose run test-ci npm run wdio -- --suite suite3"
-    -
-      script:
-        - "travis_wait docker-compose run test-ci npm run wdio -- --suite suite4"
+      script: "travis_wait docker-compose run -e DANGER_GITHUB_API_TOKEN=$DANGER_GITHUB_API_TOKEN -e TRAVIS=$TRAVIS -e CI=$CI -e HAS_JOSH_K_SEAL_OF_APPROVAL=$HAS_JOSH_K_SEAL_OF_APPROVAL -e TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST -e TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG test-ci bash -c 'npm run danger && npm run lint && npm run jest'"
+    - stage: compile - default theme
+      script: travis_wait docker-compose run test-ci npm run compile:prod
+    - stage: wdio - default theme
+      name: tiny
+      script: travis_wait docker-compose run test-ci npm run tt-wdio -- --formFactors ['tiny']
+    - script: travis_wait docker-compose run test-ci npm run tt-wdio -- --formFactors ['small']
+      name: small
+    - script: travis_wait docker-compose run test-ci npm run tt-wdio -- --formFactors ['medium']
+      name: medium
+    - script: travis_wait docker-compose run test-ci npm run tt-wdio -- --formFactors ['large']
+      name: large
+    - script: travis_wait docker-compose run test-ci npm run tt-wdio -- --formFactors ['enormous']
+      name: enormous
+    - stage: compile - fusion theme
+      script: travis_wait docker-compose run test-ci npm run compile:fusion
+    - stage: wdio - fusion theme
+      name: tiny
```

Comment:
nit: how come the tiny form factor script run is ordered `name` then `script` but the other form factors are the other way around?

Reply:
because I copied this example: https://docs.travis-ci.com/user/build-stages/#naming-your-jobs-within-build-stages
We could experiment with moving the names around. I really hate debugging and testing travis builds though because the only way to test is to push a million updates.